### PR TITLE
Warnings and fixes

### PR DESCRIPTION
**Pull request by Sabr0sa**
Hi,

I fixed a few warnings and a bug in opening the F11 menu. If the player pressed Ctrl at any point in time before pressing F11, the advanced menu would open, but I think it should only open if the player presses F11 while holding the Ctrl button. Also I activated saving of the sight range settings when in GMP mode, because I can't see the reason it was disabled - you could change the values anyway, they just weren't adopted in the next session.

### CODE REVIEW

File: D3D11Engine/GothicMemoryLocations.h [hunk 1]
```diff
@@ -27,11 +27,12 @@ static void PatchAddr( unsigned int adr, const T( &v )[n] ) {
 }
 
 // -- call macro from GothicX (thx, Zerxes!)
-#define XCALL(uAddr)                    \
-        __asm { mov esp, ebp    }       \
-        __asm { pop ebp                 }       \
-        __asm { mov eax, uAddr  }       \
-        __asm { jmp eax                 }
+#define XCALL(address)      \
```
Comment (Kirides) [inline D3D11Engine/GothicMemoryLocations.h:30]: Why was this changed?

File: D3D11Engine/GothicAPI.cpp [hunk 2]
```diff
@@ -653,12 +653,10 @@ void GothicAPI::LoadRendererWorldSettings( GothicRendererSettings& s ) {
         GetPrivateProfileIntA( "Atmoshpere", "FogColorModB", (int)(s.FogColorMod.z * 255.0f), ini.c_str() )
     );
 
-    if ( !GMPModeActive ) {
```
Comment (Kirides) [inline D3D11Engine/GothicAPI.cpp:656]: Owners and players of GMP servers wanted to have the limit.

This should also have been enforced in the F11 menu. Had to hand merge a few PRs, maybe one of them deleted the check in F11 and when starting the game.

File: D3D11Engine/GothicMemoryLocations.h (hunk 1 above)
Comment (Sabr0sa) [reply D3D11Engine/GothicMemoryLocations.h:30]: MSVC would warn in `pop ebp` that modifying the EBP register is not safe, also `leave` is an optimization over the hand-coded `mov esp, ebp; pop ebp`.

File: D3D11Engine/GothicAPI.cpp (hunk 2 above)
Comment (Sabr0sa) [reply D3D11Engine/GothicAPI.cpp:656]: I see no benefit in introducing this limit, because you can't see other players nevertheless if the GMP doesn't stream them. And without the renderer you could change the limit with the Systempack anyway.

Comment (Kirides) [reply D3D11Engine/GothicAPI.cpp:656]: Fair enough. I don't know anything about GMP, just what the people tell me. If GMP really streams people and items only if they are in the "visible range" (defined by GMP?) and that works with Systempack scaling etc. then that's fine.

Comment (Sabr0sa) [reply D3D11Engine/GothicAPI.cpp:656]: Well, I'm the developer of the GMP :D The distance to which other players are streamed can be set by the servers, but items are indeed always visible. But I don't think that's a problem.

**Comment by Kirides** [comment]
LGTM
